### ISSUE TITLE
Dosen't start program (black screen after boot logo)

### ISSUE THREAD

**Issue by lexxus-lex** [shashlik/shashlik-build#2]
My log:

/opt/shashlik/bin/shashlik-run ru.yandex.yandexbus "Транспорт"
starting emulator
error: device '(null)' not found
- waiting for device -
  emulator: WARNING: system partition size adjusted to match image file (750 MB > 200 MB)

emulator: WARNING: data partition size adjusted to match image file (750 MB > 200 MB)

setting icon3 
setting icon1 
setting icon
Creating filesystem with parameters:
    Size: 69206016
    Block size: 4096
    Blocks per group: 32768
    Inodes per group: 4224
    Inode size: 256
    Journal blocks: 1024
    Label: 
    Blocks: 16896
    Block groups: 1
    Reserved block group size: 7
Created filesystem with 11/4224 inodes and 1302/16896 blocks

And nothing, i see black screen.
Use Kubuntu 15.10 x86-64, plasma 5, nvidia driver 361.28

With flappy bird this bug too

sorry for my eng :))

**Comment by licaon-kter** [comment]
/duplicate of https://github.com/shashlik/shashlik-build/issues/1

**Comment by wtermini** [comment]
I am having the same issue except its very intermittent.  When I launch an app it opens every other time or so.  I cant find any clear error message for when it fails.  

When it fails it just seems to hang at a black screen.  I ran adb logcat and the emulator was still running and responding to touch screen events despite the black display. 

The app I tested was flappy bird and I am running Arch Linux with Plasma 5.

**Comment by andresgabrielrc** [comment]
Installed PicArts and Slack. Both "start" whit the logo and then black screen and some huge memory usage
Some specs in the screenshot
(Sorry bad english) 

![screenshot_2016-02-23_10-07-48](https://cloud.githubusercontent.com/assets/17430665/13252596/583c08a4-da16-11e5-9f6f-41ee7ae97a8c.png)

**Comment by licaon-kter** [comment]
@andresgabrielrc 
Edit `/opt/.../shashlik-run` around here: https://github.com/shashlik/shashlik-runtime-env/blob/master/scripts/shashlik-run#L53 add
`"-debug","all",` ( watch out for the commas ;))

Retry... guess you'll see [this ](https://github.com/shashlik/shashlik-build/issues/1#issuecomment-187301946)

**Comment by andresgabrielrc** [comment]
@licaon-kter 

#This is the console output

> [nne@nne-nb ~]$ /opt/shashlik/bin/shashlik-run com.Slack "Slack" > ~/log.txt
> Exception in thread Thread-1:
> Traceback (most recent call last):
>   File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
>     self.run()
>   File "/usr/lib/python3.5/threading.py", line 862, in run
>     self._target(_self._args, *_self._kwargs)
>   File "/opt/shashlik/bin/shashlik-run", line 71, in send_icon
>     os.remove(socket_path)
> PermissionError: [Errno 1] Operation not permitted: '/tmp/shashlik_controller'
> error: device '(null)' not found
> waiting for device
> error: device '(null)' not found
> waiting for device
> Killed

#This is the log file [https://mega.nz/#!cRI12LyA!TN-zTrj8l0NAyUO8PamWV2lemKZnnvgM9LKCWPeXsyI](url)

> emulator: ERROR: Invalid -datadir directory: /home/nne/.local/share/shashlik/system
> 
> emulator: Error while connecting to socket '127.0.0.1:1970': 111 -> Connection refused
> emulator: ASC 127.0.0.1:1970: Retrying connection. Connector FD = 35
> emulator: Error while connecting to socket '127.0.0.1:1970': 111 -> Connection refused
> emulator: ASC 127.0.0.1:1970: Retrying connection. Connector FD = 33
> emulator: Pend 31 bytes received from ADB host 0x140df80(so=38): CNXN\x00\x00\x00\x01\x00\x10\x00\x00\x07\x00\x00\x002\x02\x00\x00\xbc\xb1\xa7\xb1host::\x00
> emulator: Error while connecting to socket '127.0.0.1:1970': 111 -> Connection refused
> emulator: ASC 127.0.0.1:1970: Retrying connection. Connector FD = 35
> emulator: Error while connecting to socket '127.0.0.1:1970': 111 -> Connection refused
> emulator: ASC 127.0.0.1:1970: Retrying connection. Connector FD = 33

*\* First time that i do this (debugin and review the log) sorry if is not what you need and my english :D **

**Comment by licaon-kter** [comment]
`PermissionError: [Errno 1] Operation not permitted: '/tmp/shashlik_controller'`

You started as root or other user at least once? Remove that file as root as needed.

For the other error see my 2 ideas (BIOS/telnet) in issue #1 

Also, Slack has no x86 (or otherwise) libs afaics, try an app with x86 libs too (look into the APK since it's just a ZIP file).

**Comment by tristan-k** [comment]
Same here:

```
$ /opt/shashlik/bin/shashlik-run eu.siacs.conversations "Conversations"
starting emulator
emulator: WARNING: system partition size adjusted to match image file (750 MB > 200 MB)

emulator: WARNING: data partition size adjusted to match image file (750 MB > 200 MB)

setting icon3 
setting icon1 
setting icon
error: device '(null)' not found
- waiting for device -
Creating filesystem with parameters:
    Size: 69206016
    Block size: 4096
    Blocks per group: 32768
    Inodes per group: 4224
    Inode size: 256
    Journal blocks: 1024
    Label: 
    Blocks: 16896
    Block groups: 1
    Reserved block group size: 7
Created filesystem with 11/4224 inodes and 1302/16896 blocks
- waiting for device -
```

**Comment by ghost** [comment]
same error here
http://pastebin.com/zVPdx6zG

**Comment by saligari** [comment]
I have the same problem as @andresgabrielrc with flappy birds. However the first time I tried running flappy birds it started out fine but wouldn't accept any input (keyboard, mouse). After that I closed it and re-opened it and get a black screen (with the android buttons on the bottom); all subsequent tries result in the same black screen and huge memory usage.

Edit: Grammar

**Comment by vamanea** [comment]
I've had the same problem as @T3RM1NU5 above: first install everything works fine, then on restart I just get a black screen. No clear error messages in qemu logs. In adb shell I checked and the application was running fine, just didn't seem to start.

What I did was to:

```
adb shell rm -rf "/data/system/*"
```

Then after I rebooted the application started again. Seeing this I wrote a small script that runs before shashlik-run to delete everything from /data/system/\* and... everything works fine, for every app all the time!

**Comment by sandeshbhusal** [comment]
@vamanea  could you post that script please? Thanks. :)

**Comment by vamanea** [comment]
Sure, it's nothing special really.

I added this line to my fstab:

```
/home/<my user here>/.local/share/shashlik/userdata.img   /tmp/userdata/  ext4    loop,defaults,noauto,user       0       0
```

and I run this script before shashlik-run:

```
#!/bin/bash

mkdir -p /tmp/userdata
mount /tmp/userdata
rm -rf /tmp/userdata/system/*
umount /tmp/userdata
```

This way I get to do it without any password prompts.

**Comment by auhsoJyrraH** [comment]
it is not starting WhatsApp... it keeps on displaying an error message "WhatsApp has stopped"

**Comment by auhsoJyrraH** [comment]
what are the keyboard keys for playing games? and how do you make games run faster?

**Comment by xweiwei** [comment]
I am having the same issue

**Comment by DoraemonYu** [comment]
I am having the same issue  on the newest version now (0.9.3).

And I append "-debug all","-show-kernel","-verbose" arg to L88 of **shashlik-run**, then in the log I saw there was **so many these message**:
`emulator: Error while connecting to socket '127.0.0.1:1970': 111 -> Connection refused`
`emulator: ASC 127.0.0.1:1970: Retrying connection. Connector FD = XXX`  (XXX is diferent number)

other error messages is :
`ACPI Error: A valid RSDP was not found (20120320/tbxfroot-219)`
`[drm:i915_init] *ERROR* drm/i915 can't work without intel_agp module!`
`QEMU Pipe Device:rw, wait_event error`
`segfault at bfccbbdc ip b70f5e60 sp bfc8b8c0 error 4 in libcrypto.so`

**Comment by perdigao1** [comment]
I am also hing issues to run applications.
Adding
`,"-debug","all","-show-kernel","-verbose",`
to line 88, and running a simple game called kapsul infinite, it shows the logo of the application at the start but then it loops giving the following error (partially copied here)
```
emulator: ASC 127.0.0.1:1970: Retrying connection. Connector FD = 26
emulator: Error while connecting to socket '127.0.0.1:1970': 111 -> Connection refused
emulator: ASC 127.0.0.1:1970: Retrying connection. Connector FD = 28
emulator: Error while connecting to socket '127.0.0.1:1970': 111 -> Connection refused
emulator: ASC 127.0.0.1:1970: Retrying connection. Connector FD = 26
emulator: Error while connecting to socket '127.0.0.1:1970': 111 -> Connection refused
emulator: ASC 127.0.0.1:1970: Retrying connection. Connector FD = 28
emulator: Error while connecting to socket '127.0.0.1:1970': 111 -> Connection refused

```

**Comment by elvisAR-git** [comment]
just getting black screen, no soft buttons on the bottom. no response

**Comment by ahmedmujtabanizamani** [comment]
> I've had the same problem as @T3RM1NU5 above: first install everything works fine, then on restart I just get a black screen. No clear error messages in qemu logs. In adb shell I checked and the application was running fine, just didn't seem to start.
> 
> What I did was to:
> 
> ```
> adb shell rm -rf "/data/system/*"
> ```
> 
> Then after I rebooted the application started again. Seeing this I wrote a small script that runs before shashlik-run to delete everything from /data/system/\* and... everything works fine, for every app all the time!
> 

**Thanks alot bro u save my day :***

**Comment by bodqhrohro** [comment]
>`rm -rf /tmp/userdata/system/*`

This is too harsh, deleting `/tmp/userdata/system/appops.xml` is enough.

Maybe even a certain parameter in the file, I didn't investigate deeper yet.
